internal/resources/resource_host.go

```diff
@@ -431,6 +454,37 @@ func resourceMetalHostCreate(d *schema.ResourceData, meta interface{}) (err erro
 	if err != nil {
 		return err
 	}
+
+	// host create is asynchronous in Metal svc. Wait until host state is Ready.
+	createStateConf := &resource.StateChangeConf{
+		Pending: []string{
+			hStateNew,
+			hStateImagingPrep,
+			hStateImaging,
+			hStateConnecting,
+			hStateBooting,
+		},
+		Target: []string{
+			hStateReady,
+		},
+		Refresh: func() (interface{}, string, error) {
+			host, _, err := p.Client.HostsApi.GetByID(ctx, h.ID)
+			if err != nil {
+				return nil, "", fmt.Errorf("get host %v", h.ID)
+			}
+
+			return host, string(host.State), nil
+		},
+		Timeout:    d.Timeout(schema.TimeoutCreate),
+		Delay:      thirty * time.Second,
+		MinTimeout: ten * time.Second,
```

Comment:
do you mean seconds here? If we need btoh tenMinutes and tenSeconds delcare more constants with those names

Reply:
MinTimeout (defined in StateChangeConf) is the minimum timeout which is for the refresh frequency; 10 seconds is what was intended.

Reply:
> MinTimeout (defined in StateChangeConf) is the minimum timeout which is for the refresh frequency; 10 seconds is what was intended.

I think I see Mark's question. MinTimeout sounds like "Timeout in Minutes", not "MinimumTimeout". If this is for refresh frequency, why not call it RefreshInterval or RefreshTimeout instead of MinTimeout?

Reply:
MinTimeout is defined in StateChangeConf, which is part of Terraform, not something I created.